types/src/error.rs

```diff
@@ -80,6 +80,36 @@ impl<'a> fmt::Display for ErrorResponse<'a> {
 	}
 }
 
+/// The return type of the subscription's method for the rpc server implementation.
+///
+/// **Note**: The error does not contain any data and is discarded on drop.
+pub type SubscriptionResult = Result<(), SubscriptionEmptyError>;
+
+/// The error returned by the subscription's method for the rpc server implementation.
+///
+/// It contains no data, and neither is the error utilized. It provides an abstraction to make the
+/// API more ergonomic while handling errors that may occur during the subscription callback.
+#[derive(Debug)]
+pub struct SubscriptionEmptyError;
+
+impl From<anyhow::Error> for SubscriptionEmptyError {
```

Comment:
These From impls could perhaps just be:

```rust
impl <T> From<T> for SubscriptionEmptyError {
    fn from(e: T) -> Self {
        SubscriptionEmptyError
    }
}
```

Since we're not super specific anyway about the types of error that can be converted into this, maybe that's easier?

Reply:
Unfortunately, that leads to an error of `conflicting implementations of trait std::convert::From`, which is conflicting with std implementation.

Reply:
Does `impl <T> Into<SubscriptionEmptyError> for T` work or is that the same issue?

Reply:
That breaks the orphan rule with https://doc.rust-lang.org/stable/error-index.html#E0210